runtime/compiler/runtime/JITServerAOTCache.cpp

```diff
@@ -810,6 +810,12 @@ JITServerAOTCache::getClassLoaderRecord(const uint8_t *name, size_t nameLength)
 const AOTCacheClassRecord *
 JITServerAOTCache::getClassRecord(const AOTCacheClassLoaderRecord *classLoaderRecord, const J9ROMClass *romClass)
    {
+   // The current implementation cannot handle array ROM classes  - the name must be recorded
+   // properly and the hash of the class must incorporate the array ROM class, the arity of the array,
+   // and the element ROM class.
+   if (J9ROMCLASS_IS_ARRAY(romClass))
```

Comment:
Excluding primitive array classes may be overkill. Their ROMClasses store their proper names which can be used to look them up at the client.

Reply:
I was worried that the `romClass` might be from an array class with primitive element type and arity greater than 1. I think that would result in a `romClass` with a name that looks like `[I` or something, but wouldn't have any arity information. Is that an issue?

Reply:
I suspect (but you might want to check) that multidimensional primitive array classes still use the `[L` ROMClass since they are arrays of references. By primitive arrays I meant specifically 1D arrays like `[I` or `[Z`, they use properly named ROMClasses. They are not stored in the SCC though, so I guess excluding them here should result in the same behaviour as before with the underlying local SCC.

Reply:
I think I'll leave it as-is just to keep the same behaviour as before when not ignoring the SCC.